mdp/hinet/flownode.py

```diff
@@ -142,7 +144,7 @@ def get_train_function(_i_node, _node):
             # the nodes in front of the current nodes.
             # using nested scopes here instead of default args, see pep-0227
             def _train(x, *args, **kwargs):
-                if i_node > 0:
+                if _i_node > 0:
```

Comment:
Seems like this was indeed a bug, though it would only affect the efficiency of the code. I wonder if we need better testing for this method.